### PR TITLE
Type and font updates

### LINKED ISSUE #5661: Type and font changes to align with marketing updates
## Is your feature request related to a problem?

There are some [upcoming changes](https://github.com/PostHog/posthog.com/pull/1824) to our marketing website that we can roll into the product to better align the two assets. Specifically, the text styles for posthog.com are changing substantially. While we can't host these fonts as part of the product, we can replicate the styling with some other updates.

## Describe the solution you'd like

Remove GoshaSans and replace it with a font similar to Matter. Make some other updates to better align the two assets (background colors and other updates in the branding guidelines). 

## Describe alternatives you've considered

Leaving things alone until we make broader changes to the product [information architecture.](https://github.com/PostHog/posthog/issues/5346).

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

Closes #5661. Get rid of Gosha Sans and adds Inter as fallback (see issue for details). In addition, we get rid of invalid weight values. @clarkus / @corywatilo would it be worth adding support for an intermediary weight (i.e. 500/600)?

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

File: frontend/src/layout.ejs [hunk 1]
```diff
@@ -1,122 +1,121 @@
...
-  border-radius: 0 0 0.25rem 0.25rem;
-}
+            .form-signin {
+                width: 100%;
+                max-width: 400px;
+                padding: 15px;
+                margin: auto;
+            }
+            .form-signin .checkbox {
+                font-weight: normal;
```
Comment (paolodamico) [inline frontend/src/layout.ejs:42]: Only "material" change in this file, mostly ignore changes to this file.

**Comment by paolodamico** [comment]
I assume we'll be able to get rid of Good Sans too in the signup page, but that's for a separate PR

**Review by clarkus — APPROVED** [review]
Typically the weights I'm using in components are regular, semi-bold (500-ish depending on the font), and with this new update, headers will go full bold. Given that we're relying on system fonts, I don't think we need to be too concerned about character sets and such for product, and we can rely on keyword values for font-weights. We might get more precise in the future, but for now this approach makes sense to me.

File: frontend/src/scenes/funnels/Funnel.scss [hunk 2]
```diff
@@ -3,13 +3,13 @@
...
     h2.funnels-empty-state__title {
-        font-weight: 600;
+        font-weight: bold;
         font-size: 1.5rem;
         line-height: 1.6rem;
     }
 
     .funnels-empty-state__description {
-        font-weight: 500;
+        font-weight: normal;
```
Comment (rcmarron) [inline frontend/src/scenes/funnels/Funnel.scss:12]: I don't think this is needed. Looks like `font-weight: normal;` is the default here.

File: frontend/src/scenes/sessions/SessionRecordingsButton.scss [hunk 3]
```diff
@@ -33,7 +33,7 @@
 }
 
 .session-recordings-popup__label {
-    font-weight: 500;
+    font-weight: normal;
```
Comment (rcmarron) [inline frontend/src/scenes/sessions/SessionRecordingsButton.scss:36]: Same here. Looks like `font-weight: normal;` is also the default.

**Review by rcmarron — APPROVED** [review]
LGTM. 2 super small nits

**Comment by paolodamico** [comment]
Thanks @rcmarron & @clarkus for the review!

**Pull request merged**